projects/extension/sql/idempotent/001-openai.sql

```diff
@@ -242,7 +242,7 @@ as $python$
         logit_bias=ai.openai.str_arg_to_dict(logit_bias),
         logprobs=logprobs,
         top_logprobs=top_logprobs,
-        max_tokens=max_tokens,
+        max_completion_tokens=max_tokens,
```

Comment:
Do the old models still take max_tokens? Should we provide both as parameters to our function?

Reply:
`max_completion_tokens` is retrocompatible with all models. No need to pass `max_tokens` anymore.

Reply:
@jgpruitt There is the need to use one or another param based on the model; for reasoning models `max_completion_tokens` should be used, `max_token` otherwise.

I used a regex to match reasoning models, if you feel there is a better alternative please share.
I discarded the following matching approaches:

- Matching literal `o1`, `o1-mini`, `o3-mini` because whenever a new reasoning o* model goes out, it won't be supported.
- Reverse matching previous models. The list is so big and there seems to be variants of those models (including datetimes in the slug name).